core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/opendistro_for_elasticsearch/upgrade-elasticsearch.yml

```diff
@@ -1,134 +1,212 @@
 ---
-- name: Open Distro for Elasticsearch | Create upgrade state file
+- name: ODFE | Create upgrade state file
   become: true
   file:
     path: "{{ opendistro_for_elasticsearch.upgrade_state_file_path }}"
     state: touch
     mode: u=rw,g=r,o=
 
-- name: Open Distro for Elasticsearch | Ensure elasticsearch service is running
+- name: ODFE | Ensure elasticsearch service is running
   systemd:
     name: elasticsearch
     enabled: yes
     state: started
   register: elasticsearch_state
 
-- name: Open Distro for Elasticsearch | Wait for elasticsearch service to start up
+# TODO: Get host from /etc/elasticsearch/elasticsearch.yml
+- name: ODFE | Wait for elasticsearch service to start up
+  when: elasticsearch_state.changed == true
   wait_for:
     port: "{{ es_transport_port }}"
     host: "{{ ansible_default_ipv4.address | default(ansible_all_ipv4_addresses[0]) }}"
-    delay: 10
-  when: elasticsearch_state.changed == true
+    delay: 5
```

Comment:
We should add follow up task to add here check for service status instead of delay. This also relates to multiple other delays in code.

Reply:
The `wait_for` task in this particular case is ok since systemd's status `running` often only means that process started (it depends on service type) and port may be not opened yet. Delay 5 is for waiting 5s before the first check, ES service starts up quite long. But agree that this (all places) can be improved.